testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -313,7 +317,30 @@ impl RunLoop {
         // prepare to fetch the first reward cycle!
         target_burnchain_block_height = burnchain_height + pox_constants.reward_cycle_length as u64;
 
+        let should_keep_running = Arc::new(AtomicBool::new(true));
+        let keep_running_writer = should_keep_running.clone();
+
+        termination::set_handler(move || {
```

Comment:
I would take a second look at where this is getting placed.  I think you'll want to do this _before_ you start any threads.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2433/commits/b2cc9e3b1fb15402af9191467a5197a3a160c69d